src/xg.cpp

```diff
@@ -773,29 +977,38 @@ void XG::build(vector<pair<id_t, string> >& node_label,
         // write the edges in id-based format
         // we will next convert these into relative format
         for (auto end : { false, true }) {
-            auto& to_sides = to_from[make_side(n.id(), end)];
-            for (auto& e : to_sides) {
+            auto& s_sides = start_side[make_side(n.id(), end)];
+            for (auto& e : s_sides) {
                 g_iv[g++] = side_id(e);
-                g_iv[g++] = edge_type(side_is_end(e), end);
+                if (e<0){
+                    g_iv[g++] = 0;
+                }
+                else {
+                    g_iv[g++] = 1;
+                }
                 ++to_edge_count;
             }
         }
         g_iv[to_edge_count_idx] = to_edge_count;
         for (auto end : { false, true }) {
-            auto& from_sides = from_to[make_side(n.id(), end)];
-            for (auto& e : from_sides) {
+            auto& e_sides = end_side[make_side(n.id(), end)];
+            for (auto& e : e_sides) {
                 g_iv[g++] = side_id(e);
-                g_iv[g++] = edge_type(end, side_is_end(e));
+                if (e<0){
+                    g_iv[g++] = 0;
```

Comment:
Rather than having these 0/1 constants floating around and counting on future coders to just understand their semantics, it would be better to define the edge types in the header file. You could add something along lines of 
```
const static int64_t G_TO_START_EDGE_TYPE = 0;
const static int64_t G_TO_END_EDGE_TYPE = 1;
```
into the `xg::XG` namespace. You can look at the `G_EDGE_LENGTH`, etc. constants as an example.